corehq/messaging/smsbackends/telerivet/models.py

```diff
@@ -76,6 +78,20 @@ def send(self, msg, *args, **kwargs):
             'content': msg.text,
             'message_type': MESSAGE_TYPE_SMS,
         }
+
+        # This can maybe happen higher up in the stream?
+        phone_number_id = msg.phone_number.replace('+', '')
+        related_case = get_case_by_identifier(msg.domain, phone_number_id)
```

Comment:
I included the functionality to check whether the message that's being sent relates to a case here for now, but please feel free to comment on better places to rather put it.

Reply:
Another thing: 
At the moment the status callback will happen **only** when there's a case related to the message, but we could change it....and maybe it should...